gwctl/pkg/printer/httproutes_test.go

```diff
@@ -19,21 +19,205 @@ package printer
+			},
+			Spec: gatewayv1.HTTPRouteSpec{
+				Hostnames: []gatewayv1.Hostname{"example.com"},
+				CommonRouteSpec: gatewayv1.CommonRouteSpec{
+					ParentRefs: []gatewayv1.ParentReference{
+						{
+							Kind:  common.PtrTo(gatewayv1.Kind("Gateway")),
+							Group: common.PtrTo(gatewayv1.Group("gateway.networking.k8s.io")),
+							Name:  "demo-gateway-1",
+						},
+						{
+							Kind:  common.PtrTo(gatewayv1.Kind("Gateway")),
+							Group: common.PtrTo(gatewayv1.Group("gateway.networking.k8s.io")),
+							Name:  "demo-gateway-200",
+						},
+					},
+				},
+			},
+		},
+		&gatewayv1.HTTPRoute{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "bar-route-21",
+				Namespace: "ns1",
+				CreationTimestamp: metav1.Time{
+					Time: fakeClock.Now().Add(-9 * time.Hour),
+				},
+			},
+			Spec: gatewayv1.HTTPRouteSpec{
+				Hostnames: []gatewayv1.Hostname{"foo.com", "bar.com", "example.com", "example2.com", "example3.com", "example4.com", "example5.com"},
+				CommonRouteSpec: gatewayv1.CommonRouteSpec{
+					ParentRefs: []gatewayv1.ParentReference{
+						{
+							Kind:      common.PtrTo(gatewayv1.Kind("Gateway")),
+							Group:     common.PtrTo(gatewayv1.Group("gateway.networking.k8s.io")),
+							Namespace: common.PtrTo(gatewayv1.Namespace("default")),
+							Name:      "demo-gateway-200",
+						},
+					},
+				},
+			},
+		},
+		&gatewayv1.HTTPRoute{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "bax-httproute-18777",
+				Namespace: "ns2",
+				CreationTimestamp: metav1.Time{
+					Time: fakeClock.Now().Add(-5 * time.Minute),
+				},
+			},
+			Spec: gatewayv1.HTTPRouteSpec{
+				CommonRouteSpec: gatewayv1.CommonRouteSpec{
+					ParentRefs: []gatewayv1.ParentReference{
+						{
+							Kind:      common.PtrTo(gatewayv1.Kind("Gateway")),
+							Group:     common.PtrTo(gatewayv1.Group("gateway.networking.k8s.io")),
+							Namespace: common.PtrTo(gatewayv1.Namespace("ns1")),
+							Name:      "demo-gateway-345",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	params := utils.MustParamsForTest(t, common.MustClientsForTest(t, objects...))
+	discoverer := resourcediscovery.Discoverer{
+		K8sClients:    params.K8sClients,
+		PolicyManager: params.PolicyManager,
+	}
+	resourceModel, err := discoverer.DiscoverResourcesForHTTPRoute(resourcediscovery.Filter{})
+	if err != nil {
+		t.Fatalf("Failed to construct resourceModel: %v", resourceModel)
+	}
+
+	hp := &HTTPRoutesPrinter{
+		Out:   params.Out,
+		Clock: fakeClock,
+	}
+
+	hp.Print(resourceModel)
+
+	got := params.Out.(*bytes.Buffer).String()
+	want := `
+NAMESPACE  NAME                 HOSTNAMES                          PARENT REFS  AGE
+default    foo-httproute-1      example.com,example2.com + 1 more  1            24h
+default    qmn-httproute-100    example.com                        2            11h
+ns1        bar-route-21         foo.com,bar.com + 5 more           1            9h
+ns2        bax-httproute-18777  None                               1            5m
```

Comment:
That is a good opportunity to provide more deterministic outputs -- which in addition to helping with the tests should provide a better overall UX as well.

For now, it seems better to perform the sorting on the client side (i.e. in the printer). Maybe simply create a new slice of *HTTPRouteNode, sort them (by namespace breaking ties by name) and then iterate over the slice to print?

Reply:
> That is a good opportunity to provide more deterministic outputs -- which in addition to helping with the tests should provide a better overall UX as well.

@gauravkghildiyal I totally agree with you. It would be great if we could add a `-- sort-by` option later.

Reply:
> For now, it seems better to perform the sorting on the client side (i.e. in the printer). Maybe simply create a new slice of *HTTPRouteNode, sort them (by namespace breaking ties by name) and then iterate over the slice to print?

@gauravkghildiyal We should do this for the `gwctl get gateways` as well, so I'll create a new PR.